logging-z-6.2.2 release notes

theashiot opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.18, 4.17, 4.16
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview: https://92988--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/logging/logging-6.2/log6x-release-notes-6.2.html
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

theashiot committed:
logging-z-6.2.2 release notes

theashiot commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,36 @@
+This release includes 
+//link:https://access.redhat.com/errata/RHBA-2025:3908[RHBA-2025:3908].
+
+[id="logging-release-notes-6-2-2-enhancements_{context}"]
+== Enhancements
+
+* Before this update, the Cloudwatch output supported log events up to 256 KB in size. With this update, the Cloudwatch output supports up to 1 MB in size to match the updates published by Amazon Web Services (AWS). (link:https://issues.redhat.com/browse/LOG-7013[LOG-7013])
```
done

theashiot commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,36 @@
+
+////
+[id="logging-release-notes-6-2-2-cves_{context}"]
+== CVEs
```
yeah, i double-checked to be sure, didn't find any.

theashiot commented on observability/logging/logging-6.2/log6x-release-notes-6.2.adoc:
```diff
@@ -5,6 +5,6 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
-
+include::modules/log6x-6-2-2-rn.adoc[leveloffset=+1]
```
yep, fixed

anpingli commented:
LGTM

theashiot commented:
Thanks, @xperimental, @anpingli!

theashiot commented:
/label peer-review-needed

eromanova97 commented:
/remove-label peer-review-needed

/label peer-review-in-progress

eromanova97 commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+
+* Before this update, the Cloudwatch output supported log events up to 256 KB in size. With this update, the Cloudwatch output supports up to 1 MB in size to match the updates published by Amazon Web Services (AWS). (link:https://issues.redhat.com/browse/LOG-7013[LOG-7013])
+
+* Before this update, auditd log messages with multiple `msg` keys caused incorrect parsing because all `msg` values were returned as an array. This behavior conflicted with the standard auditd log format, where the first `msg` field follows the structure `msg=audit(TIMESTAMP:ID)` and contains essential metadata. With this update, only the first `msg` value is used during parsing, which resolves  ...
```
Same suggestion and reasoning here as in the other PR: (https://github.com/openshift/openshift-docs/pull/93019)
```suggestion
* Before this update, `auditd` log messages with multiple `msg` keys could cause errors in collector pods, because the standard `auditd` log format expects a single `msg` field per log entry that follows the `msg=audit(TIMESTAMP:ID)` structure. With this update, only the first `msg` value is used, which resolves the issue and ensures accurate extraction of audit metadata. (link:https://issues.redhat.com/browse/LOG-7014[LOG-7014])
```

theashiot replied:
wonderful suggestion, implemented!

eromanova97 commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+
+* Before this update, auditd log messages with multiple `msg` keys caused incorrect parsing because all `msg` values were returned as an array. This behavior conflicted with the standard auditd log format, where the first `msg` field follows the structure `msg=audit(TIMESTAMP:ID)` and contains essential metadata. With this update, only the first `msg` value is used during parsing, which resolves  ...
+
+* Before this update, collector pods would enter a crash loop due to a configuration error when attempting token-based authentication with an Elasticsearch output. With this update, token authentication to Elasticsearch output produces a valid configuration. (link:https://issues.redhat.com/browse/LOG-7017[LOG-7017])
```
Same suggestion and reasoning here as in the other PR: (https://github.com/openshift/openshift-docs/pull/93019)
```suggestion
* Before this update, collector pods would enter a crash loop due to a configuration error when attempting token-based authentication with an Elasticsearch output. With this update, token authentication with an Elasticsearch output generates a valid configuration. (link:https://issues.redhat.com/browse/LOG-7017[LOG-7017])
```

theashiot replied:
done

eromanova97 commented on modules/log6x-6-2-2-rn.adoc:
Just a note here that I had issues previously in one of my PRs (it was with xrefs but still) where even if I commented the links out, it still could cause issues with the build. This is probably ok, but I wanted to mention it just in case you faced any weird errors, that maybe removing the commented out links would help :+1:

theashiot replied:
thanks, removed :)

eromanova97 commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+[id="logging-release-notes-6-2-2-bug-fixes_{context}"]
+== Bug Fixes
+
+* Before this update, logs without the `responseStatus.code` field resulted in parsing errors within the Loki distributor when using the OpenTelemetry data model. With this update, logs without the `responseStatus.code` field are parsed correctly. (link:https://issues.redhat.com/browse/LOG-7012[LOG-7012])
```
This sentence is quite hard to read, I think because of the "when using the OpenTelemetry data model" part at the end. Would something like this maybe work:

```suggestion
* Before this update, logs without the `responseStatus.code` field caused parsing errors in the Loki `distributor` component. This happened when using the OpenTelemetry data model. With this update, logs without the `responseStatus.code` field are parsed correctly. (link:https://issues.redhat.com/browse/LOG-7012[LOG-7012])
```

eromanova97 commented on observability/logging/logging-6.2/log6x-release-notes-6.2.adoc:
```diff
@@ -6,5 +6,6 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
```
I have the same comment here regarding a short introduction that I posted here in the other RN PR: https://github.com/openshift/openshift-docs/pull/93019#issuecomment-2857470833

theashiot replied:
Just to ensure that this PR makes it in time for the release, I'll implement this suggestion as part of https://issues.redhat.com/browse/OBSDOCS-1747

eromanova97 commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+* Before this update, auditd log messages with multiple `msg` keys caused incorrect parsing because all `msg` values were returned as an array. This behavior conflicted with the standard auditd log format, where the first `msg` field follows the structure `msg=audit(TIMESTAMP:ID)` and contains essential metadata. With this update, only the first `msg` value is used during parsing, which resolves  ...
+
+* Before this update, collector pods would enter a crash loop due to a configuration error when attempting token-based authentication with an Elasticsearch output. With this update, token authentication to Elasticsearch output produces a valid configuration. (link:https://issues.redhat.com/browse/LOG-7017[LOG-7017])
+
+
```
To remove one redundant line:
```suggestion

```

theashiot replied:
done

eromanova97 reviewed (commented):
Hello, I finished the review. 

I have the same comment here regarding the 4.19 as I commented in the other RN PR: "I wanted to note that the enterprise-4.19 branch is also available, in case you might need to cherry-pick this to 4.19 as well. (The docs for 6.2 seem to reside in the 4.19 branch, so that might be the case)."

eromanova97 commented:
/remove-label peer-review-in-progress

/label peer-review-done

theashiot commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+[id="logging-release-notes-6-2-2-bug-fixes_{context}"]
+== Bug Fixes
+
+* Before this update, logs without the `responseStatus.code` field resulted in parsing errors within the Loki distributor when using the OpenTelemetry data model. With this update, logs without the `responseStatus.code` field are parsed correctly. (link:https://issues.redhat.com/browse/LOG-7012[LOG-7012])
```
done

theashiot commented:
Thanks a lot @eromanova97 for the wonderful review! I've implemented the changes and added a comment about intro, which I'll implement as part of another Jira.

best,
ashwin

theashiot commented on modules/log6x-6-2-2-rn.adoc:
```diff
@@ -0,0 +1,32 @@
+[id="logging-release-notes-6-2-2_{context}"]
+= Logging 6.2.2 Release Notes
+
+This release includes link:https://access.redhat.com/errata/RHBA-2025:4526[RHBA-2025:4526].
```
That's right

theashiot commented:
/label merge-review-needed

mburke5678 commented:
/cherrypick enterprise-4.16

mburke5678 commented:
/cherrypick enterprise-4.17
